spec/services/reports/children_served_report_service_spec.rb

```diff
@@ -109,7 +106,10 @@
     end
   end
   describe "#disposable_diapers_from_kits_total" do
+    before { DatabaseCleaner.clean_with(:truncation) }
```

Comment:
Kind of hacky, but I could not find an alternative. This test is super confusing, not sure what it is doing. I kept getting 400.

Reply:
This does look like a hack. If you can't figure it out, punt it my way and I'll take a look on Friday. I'd rather not have truncation in here unless we really need it (e.g. we're testing something related to transactions).

Reply:
@elasticspoon were you getting 400 even when running the test by itself? I ran the test without that line and it passed. If you have a link to a CI failure, can you send it my way?

Reply:
That is so weird. Its failing for me but passing CI:

```console
$ > ,be rspec spec/services/reports/children_served_report_service_spec.rb:110
Run options: include {:locations=>{"./spec/services/reports/children_served_report_service_spec.rb"=>[110]}}

Randomized with seed 45332

Reports::ChildrenServedReportService
  #disposable_diapers_from_kits_total
    calculates the number of disposable diapers that have been distributed within kits (FAILED - 1)

Failures:

  1) Reports::ChildrenServedReportService#disposable_diapers_from_kits_total calculates the number of disposable diapers that have been distributed within kits
     Failure/Error: expect(service.disposable_diapers_from_kits_total).to eq(100)

       expected: 100
            got: 400

       (compared using ==)
     # ./spec/services/reports/children_served_report_service_spec.rb:139:in `block (3 levels) in <top (required)>'

Finished in 1.37 seconds (files took 4.49 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./spec/services/reports/children_served_report_service_spec.rb:110 # Reports::ChildrenServedReportService#disposable_diapers_from_kits_total calculates the number of disposable diapers that have been distributed within kits

Randomized with seed 45332

Coverage report generated for RSpec to /home/bandito/Projects/rfg-human-essentials-fork/coverage. 942 / 9369 LOC (10.05%) covered.
Stopped processing SimpleCov as a previous error not related to SimpleCov has been detected
```

It also passes if I run the full input that is being given to CI, but when I run just that one test it fails.

Reply:
So what do you want to do here? Want to try to debug the failure on your local?

Reply:
And the results should be 100 because it's five times 10 twice since the cloth diapers are not included?

Reply:
Yep!

Reply:
Actually we did some debugging during office hours, this fails on main for not just me! BUT I think we figured out the cause to hopefully I get it fixed.

We think: when kits are created they also create items + line items. Those will create a base item. The base item is the BaseItem.first. We think that if that base item has a category of diaper which 70% of them do the test fails.

Reply:
This is unrelated to my changes and is out of scope. opened an issue https://github.com/rubyforgood/human-essentials/issues/4297